csharp-mode.el

```diff
@@ -95,6 +95,43 @@
 (c-lang-defconst c-multiline-string-start-char
   csharp ?@)
 
+(c-lang-defconst c-ml-string-opener-re
+  ;; "\\(?:\\=\\|[^\"]\\)\\(?:\"\"\\)*\\(\\(\"\\)\\)\\(?:[^\"]\\|\\'\\)"
+  csharp
+  (rx
+   (seq
+    (or point (not (any "\"")))
+    (zero-or-more "\"\"")
+    (group
+     (group "\""))
+    (or (not (any "\"")) eos))))
+
+(c-lang-defconst c-ml-string-max-opener-len
+  csharp 2)
+
+(c-lang-defconst c-ml-string-max-closer-len
+  csharp 2)
+
+(c-lang-defconst c-ml-string-any-closer-re
+  ;; "\\(?:\"\"\\)*\\(\\(\"\\)\\)\\(?:[^\"]\\|\\'\\)"
+  csharp
+  (rx
+   (seq
+    (zero-or-more "\"\"")
+    (group
+     (group "\""))
+    (or (not (any "\"")) eos))))
+
+(c-lang-defconst c-ml-string-back-closer-re
+  ;; "\\(:?\\`\\|[^\"]\\)\"*"
+  csharp
+  (rx
+   (seq
+    (group
+     (or (seq (opt ":") bos)
+         (not (any "\""))))
+    (zero-or-more "\""))))
```

Comment:
Is zero or more really correct here? Shouldn't it at least be once? (Same for above)

Reply:
I thought so too, but this is what I get doing that:
<img width="334" alt="bilde" src="https://user-images.githubusercontent.com/26711805/141853712-c30bbeda-2036-4cb5-b3a9-f2939cc62f41.png">

Reply:
Righto. That’s no good. Regexps are weird, such is life. 😅